File: Homework/quoc_khanh/lesson_5/219.ContainsDuplicateII.py
from typing import List


class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        latestIndices = {}
        for numCurrentIndex, num in enumerate(nums):
            numLatestIndex = latestIndices.get(num, -1)
            if numLatestIndex >= 0:
                if numCurrentIndex - numLatestIndex <= k:
                    return True
            latestIndices[num] = numCurrentIndex
        return False

s = Solution()
s.containsNearbyDuplicate([1,2,3,1], 3)

File: Homework/Duc_Trinh/lesson-5/219-contains-duplicate-II.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function(arr, k) {
    const hm = {}
    for (let i = 0; i < arr.length; i++) {
        if (i >= k + 1) {
            hm[arr[i - k - 1]] = false
        }
        if (hm[arr[i]]) return true
        hm[arr[i]] = true
    }
    return false
};
